src/stores/notebook.ts

```diff
@@ -47,6 +47,7 @@ export type NotebookStoreState = {
       output?: NotebookOutput;
     };
   };
+  kernelId?: string;
```

Comment:
Nice way to move the non-reactive property to a reactive one. Only thing is in TypeScript I try to be specific about the types. In this case it looks like the type is always present so it could be marked as `kernelId: string` without the question mark

Also we may want to preserve the doc comment right?

Reply:
Added back the doc comment. I changed this so that the `kernelId` starts as undefined (instead of an empty string) and then gets populated after we have a real kernel ID. IMO this is more intuitive than empty string, but let me know if you prefer it as an empty string